### PR TITLE
Add BufferTags command

### PR DESCRIPTION
**Pull request by yuki-yano**
スクリーンショット 2020-02-12 0 07 40

### CODE REVIEW

File: autoload/fzf_preview/resource.vim [hunk 1]
```diff
@@ -93,6 +93,15 @@ endfunction
 
 function! fzf_preview#resource#grep(args) abort
   return  fzf_preview#converter#convert_for_fzf(systemlist(fzf_preview#command#grep_command(a:args)), 1)
```
Comment (rlisowski) [inline autoload/fzf_preview/resource.vim:95]: Missing `endfunction`

File: autoload/fzf_preview/resource.vim [hunk 2]
```diff
@@ -93,6 +93,15 @@ endfunction
...
+
+function! fzf_preview#resource#buffer_tags() abort
+  if !filereadable(expand('%'))
+    return []
+  endif
+
+  let lines = fzf_preview#converter#convert_for_fzf(systemlist(fzf_preview#command#buffer_tags_command(expand('%'))), 1)
+  let matches = map(lines, { _, line -> matchlist(line, '^\([^\t]\+\)\t\(\S\+\)\t\(\d\+\);"') })
+  return fzf_preview#converter#convert_for_fzf(map(matches, { _, m -> m[2] . ':' . m[3] . ":\t" . m[1] }), 1)
```
Comment (rlisowski) [inline autoload/fzf_preview/resource.vim]: There is no point to repeat current filepath in every line. Maybe line numer is enough?
```suggestion
  return fzf_preview#converter#convert_for_fzf(map(matches, { _, m -> m[3] . ":\t" . m[1] }), 1)
```

File: autoload/fzf_preview/resource.vim (hunk 1 above)
Comment (yuki-yano) [reply autoload/fzf_preview/resource.vim:95]: Fixed

**Comment by kristijanhusak** [comment]
@yuki-ycino it works great!
I just found small issue, it's maybe because i use universal-ctags.
For some javascript files where I do partial imports, ctags cannot handle it and shows a warning instead for that tag, something like this:
```
ctags: Warning: ignoring null tag in lib/infrastructure/payment/stripe/stripe.js(line: 1)
ctags: Warning: ignoring null tag in lib/infrastructure/payment/stripe/stripe.js(line: 10)
```
This line doesn't get properly mapped and matches return empty array.
This patch is fixing it for me.

```diff
diff --git a/autoload/fzf_preview/resource.vim b/autoload/fzf_preview/resource.vim
index dd71fa9..69314e2 100644
--- a/autoload/fzf_preview/resource.vim
+++ b/autoload/fzf_preview/resource.vim
@@ -102,6 +102,7 @@ function! fzf_preview#resource#buffer_tags() abort
 
   let lines = fzf_preview#converter#convert_for_fzf(systemlist(fzf_preview#command#buffer_tags_command(expand('%'))), 1)
   let matches = map(lines, { _, line -> matchlist(line, '^\([^\t]\+\)\t\(\S\+\)\t\(\d\+\);"\t\(.\+\)') })
+  call filter(matches, {_, b -> !empty(b)})
   let lists = map(sort(matches, { a, b -> a[3] - b[3] }), { _, m -> [m[3], m[1], m[4]] })
   return fzf_preview#converter#convert_for_fzf(map(fzf_preview#util#align_lists(lists), { _, v -> join(v, '  ') }), 1)
 endfunction

```

**Comment by yuki-yano** [comment]
Thanks
Fix 94f31b7da09dcd3c79f99ae9a21f0dbbb1978ce9

**Comment by kristijanhusak** [comment]
@yuki-ycino looks like some of your changes returned the same issue with empty matches. Can we put it back?

**Comment by yuki-yano** [comment]
Sorry
Apparently it was incorrectly fixed when refactored.
Fixed a1d8eac8f42b72186ebbe922b57f73f0b2a631c2

**Pull request merged**